examples/arkode/CXX_serial/ark_kpr_nestedmri.cpp

```diff
@@ -0,0 +1,1904 @@
+    if (check_ptr((void*)fcontrol, "SUNAdaptController_ImpGus")) return 1;
+    if (std::min(opts.k1f, opts.k2f) > -1)
+    {
+      retval = SUNAdaptController_SetParams_ImpGus(fcontrol, opts.k1f, opts.k2f);
+      if (check_flag(retval, "SUNAdaptController_SetParams_ImpGus")) return 1;
+    }
+    break;
+  case (6):
+    fcontrol = SUNAdaptController_ImExGus(sunctx);
+    if (check_ptr((void*)fcontrol, "SUNAdaptController_ImExGus")) return 1;
+    break;
+  }
+  if ((opts.bias > -1) && (opts.fcontrol > 0))
+  {
+    retval = SUNAdaptController_SetErrorBias(fcontrol, opts.bias);
+    if (check_flag(retval, "SUNAdaptController_SetErrorBias")) return 1;
+  }
+
+  // Create ERKStep (fast) integrator
+  void* inner_arkode_mem = NULL; // ARKode memory structure
+  inner_arkode_mem       = ERKStepCreate(f_f, T0, y, sunctx);
+  if (check_ptr((void*)inner_arkode_mem, "ERKStepCreate")) return 1;
+  retval = ARKodeSetOrder(inner_arkode_mem, opts.fast_order);
+  if (check_flag(retval, "ARKodeSetOrder")) return 1;
+  retval = ARKodeSStolerances(inner_arkode_mem, opts.fast_rtol, opts.atol);
+  if (check_flag(retval, "ARKodeSStolerances")) return 1;
+  if (opts.fcontrol != 0)
+  {
+    retval = ARKodeSetAdaptController(inner_arkode_mem, fcontrol);
+    if (check_flag(retval, "ARKodeSetAdaptController")) return 1;
+    if (opts.set_h0 != 0)
+    {
+      retval = ARKodeSetInitStep(inner_arkode_mem, opts.hf);
+      if (check_flag(retval, "ARKodeSetInitStep")) return 1;
+    }
+    if (opts.fast_pq == 1)
+    {
+      retval = ARKodeSetAdaptivityAdjustment(inner_arkode_mem, 0);
+      if (check_flag(retval, "ARKodeSetAdaptivityAdjustment")) return 1;
+    }
+  }
+  else
+  {
+    retval = ARKodeSetFixedStep(inner_arkode_mem, opts.hf);
+    if (check_flag(retval, "ARKodeSetFixedStep")) return 1;
+  }
+  retval = ARKodeSetAccumulatedErrorType(inner_arkode_mem, opts.faccum);
+  if (check_flag(retval, "ARKodeSetAccumulatedErrorType")) return 1;
+  retval = ARKodeSetMaxNumSteps(inner_arkode_mem, 1000000);
+  if (check_flag(retval, "ARKodeSetMaxNumSteps")) return 1;
+  retval = ARKodeSetUserData(inner_arkode_mem, (void*)&opts);
+  if (check_flag(retval, "ARKodeSetUserData")) return 1;
+
+  // Create inner stepper
+  MRIStepInnerStepper inner_stepper = NULL;
+  retval = ERKStepCreateMRIStepInnerStepper(inner_arkode_mem, &inner_stepper);
+  if (check_flag(retval, "ERKStepCreateMRIStepInnerStepper")) return 1;
+
+  // Create intermediate and slow controller objects, and select orders of accuracy as relevant
+  SUNAdaptController scontrol     = NULL;
+  SUNAdaptController scontrol_H   = NULL;
+  SUNAdaptController scontrol_Tol = NULL;
+  SUNAdaptController mcontrol     = NULL;
+  SUNAdaptController mcontrol_H   = NULL;
+  SUNAdaptController mcontrol_Tol = NULL;
+  switch (opts.scontrol)
+  {
+  case (1):
+    scontrol_H = SUNAdaptController_I(sunctx);
+    if (check_ptr((void*)scontrol_H, "SUNAdaptController_I (slow H)")) return 1;
```

Comment:
None of our other examples use braces in `switch` statements.  I added them as requested, primarily because these are "fat" blocks within each case, but if we want to enforce braces in switch statements more generally then this maybe this should be added to the formatter?

Reply:
I was only referring to the if statements only not the switch. I selected only one line for this comment but it includes a few before including the case which made this ambiguous.